anker/src/state.rs

```diff
@@ -107,22 +123,43 @@ impl Anker {
         Ok(())
     }
 
-    /// Confirm that the provided reserve account is the one that belongs to this instance.
+    /// Confirm that the provided stSOL reserve accounts is the one that
+    /// belongs to this instance.
     ///
-    /// This does not check that the reserve is an stSOL account.
-    pub fn check_reserve_address(
+    /// This does not check that the stSOL reserve is an stSOL account.
+    pub fn check_st_sol_reserve_address(
         &self,
         anker_program_id: &Pubkey,
         anker_instance: &Pubkey,
-        reserve_account_info: &AccountInfo,
+        st_sol_reserve_account_info: &AccountInfo,
     ) -> ProgramResult {
         self.check_derived_account_address(
-            "the reserve account",
-            ANKER_RESERVE_ACCOUNT,
-            self.reserve_account_bump_seed,
+            "the stSOL reserve account",
+            ANKER_STSOL_RESERVE_ACCOUNT,
+            self.st_sol_reserve_account_bump_seed,
             anker_program_id,
             anker_instance,
-            reserve_account_info,
+            st_sol_reserve_account_info,
+        )
+    }
+
+    /// Confirm that the provided UST reserve accounts is the one that
+    /// belongs to this instance.
+    ///
+    /// This does not check that the UST reserve is an UST account.
+    pub fn check_ust_reserve_address(
+        &self,
+        anker_program_id: &Pubkey,
+        anker_instance: &Pubkey,
+        ust_reserve_account_info: &AccountInfo,
+    ) -> ProgramResult {
+        self.check_derived_account_address(
+            "the UST reserve account",
+            ANKER_UST_RESERVE_ACCOUNT,
+            self.st_sol_reserve_account_bump_seed,
```

Comment:
This should be the `ust_reserve_account_bump_seed`, not `st_sol`.

Reply:
👍  Good catch!